orttraining/orttraining/training_api/include/module.h

```diff
@@ -11,8 +11,10 @@ namespace api {
 struct Parameter {
  public:
   // Create parameter
-  Parameter(std::string name, const OrtValue& data)
-      : name_(name), data_(data) {
+  Parameter(std::string name, const OrtValue& data, bool requires_grad)
+      : name_(name), data_(data), requires_grad_(requires_grad) {
+    ORT_ENFORCE(data_.IsAllocated());
+    ORT_ENFORCE(!name_.empty(), "Parameter must have a non-empty name.");
```

Comment:
on line 22 and 31 : change the return value to const std::string& and make the function const noexcept?

Reply:
i realized gradient name was not being set/used anywhere, so removed the field